sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Listeners/EventHubsScaleMonitor.cs

```diff
@@ -0,0 +1,377 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Messaging.EventHubs;
+using Azure.Messaging.EventHubs.Consumer;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
+using Microsoft.Azure.WebJobs.EventHubs.Processor;
+using Microsoft.Azure.WebJobs.Host.Scale;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.WebJobs.EventHubs.Listeners
+{
+    internal class EventHubsScaleMonitor : IScaleMonitor<EventHubsTriggerMetrics>
+    {
+        private const string EventHubContainerName = "azure-webjobs-eventhub";
```

Comment:
There's a lot to unpack here, but it is probably worth considering creating a custom processor based on `EventProcessor<TPartition>` where the bindings would have greater insight and control over the processor, as well as native batch processing.     _(It's also likely to be needed for some things that the service teams have asked for)_